src/components/Popover/index.stories.js

```diff
@@ -1,170 +1,56 @@
+      className="bgColor-white"
+      style={{ border: 'red dotted 3px', background: 'white', padding: '8px' }}
     >
       popover content
     </div>
   ),
 };
 
-export const styledPopoverContent = Template.bind({});
-styledPopoverContent.args = {
-  trigger: <Button theme="outlined" label="Open popover" />,
-  position: 'bottom',
-  alignment: 'start',
-  distance: 8,
-  children: (
-    <div className="bgColor--white rounded--all elevation--2 padding--all">
-      <h3 className="type--head3">Look at me</h3>
-      <p>
-        <em>i am the popover</em>
-      </p>
-    </div>
-  ),
-};
-
-export const controlledPopover = (args) => (
-  <State
-    initialValue={true}
-    render={(myIsOpen, setMyIsOpen) => (
-      <Popover
-        {...args}
-        trigger={
-          <button
-            onClick={() => {
-              setMyIsOpen(!myIsOpen);
-            }}
-          >
-            Open popover
-          </button>
-        }
-        position="bottom"
-        alignment="start"
-        interactionMode="controlled"
-        isOpen={myIsOpen}
-        onUserDismiss={() => {
-          setMyIsOpen(false);
-        }}
-        distance={8}
-      >
-        <div className="bgColor--white rounded--all elevation--2 padding--all">
-          <h3 className="type--head3">Look at me</h3>
-          <p>
-            <em>i am the popover</em>
-          </p>
-        </div>
-      </Popover>
-    )}
-  />
-);
-
-controlledPopover.parameters = {
-  docs: {
-    description: {
-      story:
-        'The "controlled" `interactionMode` follows the conventions of a [controlled component](https://reactjs.org/docs/forms.html#controlled-components), meaning the open state of the popover is exclusively controlled by the prop `isOpen`. To update the state of `isOpen` based on user interaction, you must use the `onUserDismiss` callback. The `onUserDismiss` function is invoked when 1. The user clicks the trigger element 2. The user presses `Escape` 3. the user clicks away from the popover. If you\'d like to disable scrolling of some element when a popover is open you can pass a ref of the scrolling element to the `disableScrollRef` prop.',
-    },
-  },
-};
-
-export const closeOnScroll = (args) => {
-  const containerEl = useRef(null);
+export const controlledPopover = (args) => {
+  const [isOpen, setIsOpen] = useState(true);
   return (
-    <div
-      ref={containerEl}
-      style={{ height: 200, border: '1px dashed red', overflow: 'scroll' }}
-    >
-      <div style={{ height: 500, background: 'yellow' }}>
-        <Popover
-          {...args}
-          trigger={<Button theme="outlined" label="Open popover" />}
-          position="bottom"
-          alignment="start"
-          distance={8}
-          closeOnScrollRef={containerEl}
+    <Popover
+      {...args}
+      onClickOutside={() => setIsOpen(false)}
+      triggerElement={
+        <button
+          onClick={() => {
+            setIsOpen(!isOpen);
```

Comment:
We could investigate as a follow up, but storybook CSF seems to be doing some "interesting" things with this code example:

<img width="688" alt="Screen Shot 2020-12-31 at 12 25 31 PM" src="https://user-images.githubusercontent.com/231252/103420014-6ef00000-4b63-11eb-9ceb-9b6fb818bf49.png">

Reply:
I think this is mostly storybook limitations.. it has trouble unpacking functions.